apps/els_lsp/src/els_general_provider.erl

```diff
@@ -86,21 +86,19 @@ handle_request({initialized, _Params}, State) ->
   els_distribution_server:start_distribution(NodeName),
   ?LOG_INFO("Started distribution for: [~p]", [NodeName]),
   case els_config:get(bsp_enabled) of
-    true ->
-      ok = els_bsp_client:start_server(RootUri),
-      {ok, Vsn} = application:get_key(erlang_ls, vsn),
-      els_bsp_client:request(
-        <<"build/initialize">>
-          , #{ <<"displayName">>  => <<"Erlang LS BSP Client">>
-             , <<"version">>      => list_to_binary(Vsn)
-             , <<"bspVersion">>   => <<"2.0.0">>
-             , <<"rootUri">>      => RootUri
-             , <<"capabilities">> => #{ <<"languageIds">> => [<<"erlang">>] }
-             , <<"data">>         => #{}
-             }
-       );
     false ->
-      ok
+      ok;
+    BspEnabled ->
+      BspStatus = els_bsp_provider:start(RootUri),
+      case {BspEnabled, BspStatus} of
```

Comment:
It is handled - if `false`, we just return `ok` always. Otherwise, try to start the server. If it starts up, great! If not, and `BspEnabled` is `true`, we log an error and kill the LS. If `BspEnabled` is something else (ostensibly `auto`) we log a warning but carry on.

Reply:
So maybe document `auto` and make it explicit in the matching?